source/simulator/introspection.cc

```diff
@@ -173,6 +173,7 @@ namespace aspect
     :
     FEVariableCollection<dim>(variable_definition),
     n_components (FEVariableCollection<dim>::n_components()),
+    n_compositional_fields (n_components-dim-2),
```

Comment:
please use ``parameters.names_of_compositional_fields.size()`` instead. I think this can break with melt and/or other additional equations otherwise.